File: dummy_creator.php
<?php

namespace persek_web_protocol;


// includes
require_once __DIR__.'/../persek_protocol/dummy_creator.php';
require_once __DIR__.'/../basic_parser/dummy_creator.php';
require_once __DIR__.'/../generic_protocol/dummy_creator.php';
require_once __DIR__.'/../basic_objects/dummy_creator.php';
require_once __DIR__.'/../dtmf_tools/dummy_creator.php';
require_once __DIR__.'/../lang_tools/dummy_creator.php';
require_once 'object_initializer.php';

// enums

function create_dummy__ReminderStatus_state_e()
{
    $res = ReminderStatus_state_e__IDLE;

    return $res;
}

// objects

function create_dummy__Reminder()
{
    $res = new Reminder;

    initialize__Reminder( $res
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__int32()
        , \basic_objects\create_dummy__LocalTime()
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__map( '\basic_parser\create_dummy__string',  '\basic_parser\create_dummy__string' ) // Map
        , \basic_parser\create_dummy__map( '\dtmf_tools\create_dummy__tone_e',  '\persek_protocol\create_dummy__ReminderAction' ) // Map
        , \persek_protocol\create_dummy__ReminderOptions()
        );
    return $res;
}

function create_dummy__ReminderStatus()
{
    $res = new ReminderStatus;

    initialize__ReminderStatus( $res
        , \basic_parser\create_dummy__int32()
        , create_dummy__ReminderStatus_state_e()
        , \basic_parser\create_dummy__int32()
        , \basic_objects\create_dummy__LocalTime()
        , \basic_parser\create_dummy__int32()
        , \basic_objects\create_dummy__LocalTime()
        , \basic_objects\create_dummy__LocalTime()
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        );
    return $res;
}

function create_dummy__Contact()
{
    $res = new Contact;

    initialize__Contact( $res
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_objects\create_dummy__Date()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__string()
        );
    return $res;
}

// messages

function create_dummy__GetReminderStatusRequest()
{
    $res = new GetReminderStatusRequest;

    initialize__GetReminderStatusRequest( $res
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__string()
        , \basic_objects\create_dummy__LocalTimeRange()
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__int32()
        , \lang_tools\create_dummy__lang_e()
        );
    return $res;
}

function create_dummy__GetReminderStatusResponse()
{
    $res = new GetReminderStatusResponse;

    initialize__GetReminderStatusResponse( $res
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__vector( '\persek_web_protocol\create_dummy__ReminderStatus' ) // Array
        );
    return $res;
}

function create_dummy__FindContactsRequest()
{
    $res = new FindContactsRequest;

    initialize__FindContactsRequest( $res
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__int32()
        , \lang_tools\create_dummy__lang_e()
        );
    return $res;
}

function create_dummy__FindContactsResponse()
{
    $res = new FindContactsResponse;

    initialize__FindContactsResponse( $res
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__vector( '\persek_web_protocol\create_dummy__Contact' ) // Array
        );
    return $res;
}

function create_dummy__GetContactRequest()
{
    $res = new GetContactRequest;

    initialize__GetContactRequest( $res
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__int32()
        , \lang_tools\create_dummy__lang_e()
        );
    return $res;
}

function create_dummy__GetContactResponse()
{
    $res = new GetContactResponse;

    initialize__GetContactResponse( $res
        , create_dummy__Contact()
        );
    return $res;
}

function create_dummy__AddReminderRequest()
{
    $res = new AddReminderRequest;

    initialize__AddReminderRequest( $res
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__int32()
        , create_dummy__Reminder()
        );
    return $res;
}

function create_dummy__AddReminderResponse()
{
    $res = new AddReminderResponse;

    initialize__AddReminderResponse( $res
        , \basic_parser\create_dummy__int32()
        );
    return $res;
}

function create_dummy__ModifyReminderRequest()
{
    $res = new ModifyReminderRequest;

    initialize__ModifyReminderRequest( $res
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__int32()
        , create_dummy__Reminder()
        );
    return $res;
}

function create_dummy__ModifyReminderResponse()
{
    $res = new ModifyReminderResponse;

    initialize__ModifyReminderResponse( $res
        );
    return $res;
}

function create_dummy__GetReminderRequest()
{
    $res = new GetReminderRequest;

    initialize__GetReminderRequest( $res
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__int32()
        );
    return $res;
}

function create_dummy__GetReminderResponse()
{
    $res = new GetReminderResponse;

    initialize__GetReminderResponse( $res
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__int32()
        , \basic_parser\create_dummy__string()
        , create_dummy__Reminder()
        );
    return $res;
}

# namespace_end persek_web_protocol


?>

File: dummy_creator.cpp
// includes
#include "dummy_creator.h"
#include "persek_protocol/dummy_creator.h"
#include "basic_parser/dummy_creator.h"
#include "generic_protocol/dummy_creator.h"
#include "basic_objects/dummy_creator.h"
#include "dtmf_tools_protocol/dummy_creator.h"
#include "lang_tools_protocol/dummy_creator.h"
#include "object_initializer.h"

namespace persek_web_protocol
{

namespace dummy
{

// enums

ReminderStatus_state_e create__ReminderStatus_state_e()
{
    static const unsigned SIZE = 5;

    static const ReminderStatus_state_e values[SIZE] = { ReminderStatus_state_e::IDLE, ReminderStatus_state_e::ACTIVE, ReminderStatus_state_e::COMPLETED_OK, ReminderStatus_state_e::COMPLETED_FAILED, ReminderStatus_state_e::WAITING_REDIAL_TIMER,  };

    auto res = values[ ::basic_parser::dummy::create__uint32() % SIZE ];

    return res;
}

// objects

Reminder create__Reminder()
{
    Reminder res;

    ::persek_web_protocol::initialize( & res
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_objects::dummy::create__LocalTime()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__map_t<std::string, std::string, std::string(*)(), std::string(*)()>( & ::basic_parser::dummy::create__string, & ::basic_parser::dummy::create__string ) // Map
        , ::basic_parser::dummy::create__map_t<dtmf_tools::tone_e, persek_protocol::ReminderAction, dtmf_tools::tone_e(*)(), persek_protocol::ReminderAction(*)()>( & ::dtmf_tools::dummy::create__tone_e, & ::persek_protocol::dummy::create__ReminderAction ) // Map
        , ::persek_protocol::dummy::create__ReminderOptions()
        );

    return res;
}

ReminderStatus create__ReminderStatus()
{
    ReminderStatus res;

    ::persek_web_protocol::initialize( & res
        , ::basic_parser::dummy::create__uint32()
        , dummy::create__ReminderStatus_state_e()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_objects::dummy::create__LocalTime()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_objects::dummy::create__LocalTime()
        , ::basic_objects::dummy::create__LocalTime()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        );

    return res;
}

Contact create__Contact()
{
    Contact res;

    ::persek_web_protocol::initialize( & res
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_objects::dummy::create__Date()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__string()
        );

    return res;
}

// messages

GetReminderStatusRequest * create__GetReminderStatusRequest()
{
    auto res = new GetReminderStatusRequest;

    ::persek_web_protocol::initialize( res
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__string()
        , ::basic_objects::dummy::create__LocalTimeRange()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__uint32()
        , ::lang_tools::dummy::create__lang_e()
        );

    return res;
}

GetReminderStatusResponse * create__GetReminderStatusResponse()
{
    auto res = new GetReminderStatusResponse;

    ::persek_web_protocol::initialize( res
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__vector_t<ReminderStatus, ReminderStatus(*)()>( & dummy::create__ReminderStatus ) // Array
        );

    return res;
}

FindContactsRequest * create__FindContactsRequest()
{
    auto res = new FindContactsRequest;

    ::persek_web_protocol::initialize( res
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__uint32()
        , ::lang_tools::dummy::create__lang_e()
        );

    return res;
}

FindContactsResponse * create__FindContactsResponse()
{
    auto res = new FindContactsResponse;

    ::persek_web_protocol::initialize( res
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__vector_t<Contact, Contact(*)()>( & dummy::create__Contact ) // Array
        );

    return res;
}

GetContactRequest * create__GetContactRequest()
{
    auto res = new GetContactRequest;

    ::persek_web_protocol::initialize( res
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__uint32()
        , ::lang_tools::dummy::create__lang_e()
        );

    return res;
}

GetContactResponse * create__GetContactResponse()
{
    auto res = new GetContactResponse;

    ::persek_web_protocol::initialize( res
        , dummy::create__Contact()
        );

    return res;
}

AddReminderRequest * create__AddReminderRequest()
{
    auto res = new AddReminderRequest;

    ::persek_web_protocol::initialize( res
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__uint32()
        , dummy::create__Reminder()
        );

    return res;
}

AddReminderResponse * create__AddReminderResponse()
{
    auto res = new AddReminderResponse;

    ::persek_web_protocol::initialize( res
        , ::basic_parser::dummy::create__uint32()
        );

    return res;
}

ModifyReminderRequest * create__ModifyReminderRequest()
{
    auto res = new ModifyReminderRequest;

    ::persek_web_protocol::initialize( res
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__uint32()
        , dummy::create__Reminder()
        );

    return res;
}

ModifyReminderResponse * create__ModifyReminderResponse()
{
    auto res = new ModifyReminderResponse;

    ::persek_web_protocol::initialize( res
        );

    return res;
}

GetReminderRequest * create__GetReminderRequest()
{
    auto res = new GetReminderRequest;

    ::persek_web_protocol::initialize( res
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__uint32()
        );

    return res;
}

GetReminderResponse * create__GetReminderResponse()
{
    auto res = new GetReminderResponse;

    ::persek_web_protocol::initialize( res
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__uint32()
        , ::basic_parser::dummy::create__string()
        , dummy::create__Reminder()
        );

    return res;
}

} // namespace dummy

} // namespace persek_web_protocol

